proxy/http/HttpTransact.h

```diff
@@ -755,8 +759,10 @@ class HttpTransact
     bool is_websocket        = false;
     bool did_upgrade_succeed = false;
 
-    // Some queue info
-    bool origin_request_queued = false;
+    // Track if this SM has reserved (incremented) the outbound connection tracking data.
```

Comment:
Seems like this should be part of the class that keeps track of state and has methods around some of the logic instead of having the state variables exposed in `HttpTransact::State` and having the logic in `HttpSM::do_http_server_open`

Reply:
Isn't `HttpTransact::State` the class that keeps track of state for the HTTP state machine?

Reply:
The new class would be a member of `HttpTransact::State`.  The new implementation, much like the pervious, has part of it scatters all over the place.

Reply:
I think that will end up worse, because of the heavy interaction with other parts of the `HttpSM`.  `http_do_server_open` is the only place with non-trivial code and that logic touches a lot of `HttpSM` internals. I'd rather have `HttpSM` do that, than have a subclass reach back.